src/transforms/geoip.rs

```diff
@@ -146,19 +186,48 @@ impl Transform for Geoip {
         // If we have any of the geoip fields missing, we insert
         // empty values so that we know that the transform was executed
         // but the lookup didn't find the result
-        let geoip_fields = [
-            format!("{}.city_name", target_field),
-            format!("{}.country_code", target_field),
-            format!("{}.continent_code", target_field),
-            format!("{}.timezone", target_field),
-            format!("{}.latitude", target_field),
-            format!("{}.longitude", target_field),
-            format!("{}.postal_code", target_field),
-        ];
-        for field in geoip_fields.iter() {
+        if self.dbreader.metadata.database_type == "GeoLite2-ASN"
+            || self.dbreader.metadata.database_type == "GeoIP2-ISP"
+        {
             let e = event.as_mut_log();
-            if e.get(&Atom::from(field.to_string())).is_none() {
-                e.insert(Atom::from(field.to_string()), Value::from(""));
+            if e.get(&Atom::from(format!(
+                "{}.autonomous_system_number",
+                target_field
+            )))
+            .is_none()
+            {
+                e.insert(
+                    Atom::from(format!("{}.autonomous_system_number", target_field)),
+                    Value::from(0),
+                );
+            }
```

Comment:
```suggestion
            e.try_insert(&Atom::from(format!("{}.autonomous_system_number", target_field)), Value::from(0));
```

Same for other `get(..).is_none()` + `insert`.